openhands/sdk/conversation/conversation.py

```diff
@@ -109,9 +109,16 @@ def send_message(self, message: str | Message) -> None:
         )
         with self.state:
             if self.state.agent_status == AgentExecutionStatus.FINISHED:
-                self.state.agent_status = (
-                    AgentExecutionStatus.IDLE
-                )  # now we have a new message
+                if getattr(self.state, "_run_in_progress", False):
```

Comment:
This feels weird -- shouldn't we not be able to hold the lock when run is in progress? 🤔 can we ask agent to double check?